util/util.go

```diff
@@ -377,6 +379,28 @@ func logIfNotErrno(err error, what string, ignores ...syscall.Errno) (logged boo
 	return true
 }
 
+// LogJSON is a helper function to log data structures as JSON.
+// The given wrappingKey will contain the provided data.
+// Unexpected characters will be replaced by asterisks.
+// The logged lines then may be post-processed with external pretty log parsers.
+func LogJSON(wrappingKey string, data interface{}) {
+	rawResult, err := json.Marshal(data)
```

Comment:
I think this function check the logging level, and if not in debug mode or lower return immediately.  Lots of processing to do nothing.